Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -0,0 +1,159 @@
+
+        public RestConnector mOctaneRestConnector;
+        public EntityService entityService;
+        protected LwssoAuthenticationStrategy lwssoAuthenticationStrategy;
+        protected WorkspaceContext workspaceContext;
+        protected SharedSpaceContext sharedSpaceContext;
+
+        public override bool ConnectALMProject()
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool ConnectALMServer()
+        {
+            try
+            {
+                string host;
+                string userName;
+                string password;
+                Reporter.ToLog(eLogLevel.DEBUG, "Connecting to qTest server");
+                mOctaneRestConnector = new RestConnector();
+                entityService = new EntityService(mOctaneRestConnector);
+                if (!mOctaneRestConnector.IsConnected())
+                {
+                    string ignoreServerCertificateValidation = "false"; //ConfigurationManager.AppSettings["ignoreServerCertificateValidation"];
+                    if (ignoreServerCertificateValidation != null && ignoreServerCertificateValidation.ToLower().Equals("true"))
+                    {
+                        NetworkSettings.IgnoreServerCertificateValidation();
+                    }
+                    NetworkSettings.EnableAllSecurityProtocols();
+
+                    host = ALMCore.DefaultAlmConfig.ALMServerURL;
+
+                    // If webAppUrl is empty we do not try to connect.
+                    if (string.IsNullOrWhiteSpace(host)) return false;
+
+                    IConnectionInfo connectionInfo;
+                    //string clientId = ConfigurationManager.AppSettings["clientId"];
+                    //if (clientId != null)
+                    //{
+                    //    userName = clientId;
+                    //    connectionInfo = new APIKeyConnectionInfo(clientId, ConfigurationManager.AppSettings["clientSecret"]);
+                    //}
+                    //else
+                    //{
+                    userName = ALMCore.DefaultAlmConfig.ALMUserName;  // "Jinendrag"; //ConfigurationManager.AppSettings["userName"];
+                    password = ALMCore.DefaultAlmConfig.ALMPassword;  //"Amdocs@123";//ConfigurationManager.AppSettings["password"];
+                        connectionInfo = new UserPassConnectionInfo(userName, password);
+                   // }
+
+                    lwssoAuthenticationStrategy = new LwssoAuthenticationStrategy(connectionInfo);
+                    var result =  Task.Run(() =>
+                    {
+                        return  mOctaneRestConnector.Connect(host, lwssoAuthenticationStrategy);
+                     
+                    });
+
+                    return result.Result;
+
+                  //  return mOctaneRestConnector.IsConnected();
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6200394)